packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -140,3 +203,74 @@ export function unstable_httpSubscriptionLink<
     };
   };
 }
+
+/**
+ * We wrap EventSource so that is can be reinitialized with new options
+ */
+class EventSourceWrapper implements Partial<EventSource> {
+  private es: EventSource;
+
+  private listeners: Partial<Record<keyof EventSourceEventMap, any[][]>> = {};
```

Comment:
looks like this could just be

```ts
  private listeners: Array<{
    type: keyof EventSourceEventMap;
    callback: (...args: any[]) => void
    options?: x
  }>
```

Reply:
Type was slightly wrong with the `any` but best to use the Record as there are known keys and that makes adding and removing event listeners more efficient later